src/auth/auth.service.ts

```diff
@@ -3,63 +3,85 @@ import { TokenPayload } from './interfaces/token-payload.interface';
+  ) {}
 
-  public register(body: AuthDTO) {
+  public async register(body: AuthDTO) {
     const clientId = uuidv4();
     const { serverId, scopes } = body;
+    let tokens: string[] = [];
+    try {
+      tokens = await this.astraService
+        .get<string[]>('tokens', serverId, 'tokens')
+        .toPromise();
+    } catch (e) {
+      tokens = [];
+    }
 
     const payload: TokenPayload = {
       clientId,
       keyspace: serverId,
       scopes,
     };
-    if (!this.configCollection[serverId]) {
-      this.configCollection[serverId] = { knownClients: [] };
-    }
-    this.configCollection[serverId].knownClients = [
-      ...this.configCollection[serverId].knownClients,
-      clientId,
-    ];
+    tokens = [...tokens, clientId];
+    await this.astraService
+      .replace('tokens', tokens, serverId, 'tokens')
+      .toPromise();
     //TODO token-expiry
     const signedToken = this.jwtService.sign(payload, { expiresIn: '1y' });
     const decoded: any = this.jwtService.decode(signedToken);
     const expiresIn: number = decoded.exp - Math.round(Date.now() / 1000);
     return { ...payload, accessToken: signedToken, expiresIn };
   }
 
-  public validateClient(payload: TokenPayload): boolean {
+  public async validateClient(payload: TokenPayload): Promise<boolean> {
     const { keyspace, clientId } = payload;
-    if (
-      this.configCollection[keyspace] &&
-      this.configCollection[keyspace].knownClients.includes(clientId)
-    ) {
+    let tokens: string[];
+    try {
+      tokens = await this.astraService
+        .get<string[]>('tokens', keyspace, 'tokens')
+        .toPromise();
+    } catch {
+      return false;
+    }
+    if (tokens && tokens.includes(clientId)) {
       return true;
     }
     return false;
   }
 
-  public removeClient(token: string) {
+  public async removeClient(token: string): Promise<boolean> {
+    let tokens: string[] = null;
     if (!token)
       throw new HttpException('Please provide token', HttpStatus.BAD_REQUEST);
 
     const decoded = this.jwtService.decode(token) as TokenPayload;
 
     try {
-      this.configCollection[
-        decoded.keyspace
-      ].knownClients = this.configCollection[
-        decoded.keyspace
-      ].knownClients.filter((client) => client !== decoded.clientId);
-      console.log(this.configCollection);
+      tokens = await this.astraService
+        .get<string[]>('tokens', decoded.keyspace, 'tokens')
+        .toPromise();
+    } catch (e) {
       return;
+    }
+
+    tokens = tokens.filter((clientID) => clientID !== decoded.clientId);
+    try {
+      await this.astraService
+        .replace('tokens', tokens, decoded.keyspace, 'tokens')
+        .toPromise();
     } catch (e) {
-      throw new HttpException('Invalid client id', HttpStatus.BAD_REQUEST);
+      throw new HttpException(
+        'Deleting Token went wrong',
+        HttpStatus.INTERNAL_SERVER_ERROR,
+      );
     }
+    return;
```

Comment:
I think we both code this function but you beat me to pushing it 😂 

I had this, it looks a bit different
```ts
public async removeClient(token: string) {
    if (!token)
      throw new HttpException('Please provide token', HttpStatus.BAD_REQUEST);

    const { clientId, keyspace } = this.jwtService.decode(
      token,
    ) as TokenPayload;

    try {
      await this.astraService.delete(clientId, keyspace, 'tokens');
    } catch (e) {
      throw new HttpException('Invalid client id', HttpStatus.BAD_REQUEST);
    }

    return;
  }
```

Reply:
I can probably ditch my changes and port over the tests and validate endpoint

Reply:
I have brought over the tests I wrote. I think the delete needs a step def to use the ID